extensions/msal-node-extensions/test/broker/NativeBrokerPlugin.spec.ts

```diff
@@ -0,0 +1,1320 @@
+                    expiresOn: testAuthenticationResult.expiresOn.getTime(),
+                    isPopAuthorization: false,
+                    account: testMsalRuntimeAccount,
+                    CheckError: () => {},
+                    telemetryData: ""
+                };
+                expect(correlationId).toEqual(testCorrelationId);
+                expect(account).toStrictEqual(testMsalRuntimeAccount);
+                callback(result);
+
+                return asyncHandle;
+            });
+
+            const nativeBrokerPlugin = new NativeBrokerPlugin();
+            const request: NativeRequest = {
+                clientId: TEST_CLIENT_ID,
+                scopes: testAuthenticationResult.scopes,
+                correlationId: testCorrelationId,
+                authority: testAuthenticationResult.authority,
+                redirectUri: TEST_REDIRECTURI,
+                prompt: PromptValue.NONE,
+                accountId: testMsalRuntimeAccount.accountId
+            }
+            const response = await nativeBrokerPlugin.acquireTokenInteractive(request);
+            expect(response).toStrictEqual<AuthenticationResult>(testAuthenticationResult);
+        });
+
+        it("Calls SignInSilentlyAsync and returns successful response if prompt is set to none and user is not signed-in", async () => {
+            const testCorrelationId = generateCorrelationId();
+            const testAuthenticationResult = getTestAuthenticationResult(testCorrelationId);
+
+            jest.spyOn(msalNodeRuntime, "SignInSilentlyAsync").mockImplementation((authParams: AuthParameters, correlationId: string, callback: (result: AuthResult) => void) => {
+                const result: AuthResult = {
+                    idToken: JSON.stringify(testAuthenticationResult.idTokenClaims),
+                    accessToken: testAuthenticationResult.accessToken,
+                    rawIdToken: testAuthenticationResult.idToken,
+                    grantedScopes: testAuthenticationResult.scopes.join(" "),
+                    expiresOn: testAuthenticationResult.expiresOn.getTime(),
+                    isPopAuthorization: false,
+                    account: testMsalRuntimeAccount,
+                    CheckError: () => {},
+                    telemetryData: ""
+                };
+                expect(correlationId).toEqual(testCorrelationId);
+                callback(result);
+
+                return asyncHandle;
+            });
+
+            const nativeBrokerPlugin = new NativeBrokerPlugin();
+            const request: NativeRequest = {
+                clientId: TEST_CLIENT_ID,
+                scopes: testAuthenticationResult.scopes,
+                correlationId: testCorrelationId,
+                authority: testAuthenticationResult.authority,
+                redirectUri: TEST_REDIRECTURI,
+                prompt: PromptValue.NONE
+            }
+            const response = await nativeBrokerPlugin.acquireTokenInteractive(request);
```

Comment:
Just want to make sure here that when prompt = none, acquireTokenInteractive request falls back to SignInSilentlyAsync request? Because the test description says "Calls SignInSilentlyAsync".

Reply:
Yes. If prompt=none then we call either SignInSilentlyAsync or AcquireTokenSilentlyAsync depending on whether or not an account was included in the request.